### PR TITLE
Aligning Atlas with LayerKit v0.16.0 Changes

### PR DESCRIPTION
**Pull request by kmahal**

### CODE REVIEW

File: Code/Controllers/ATLConversationListViewController.m [hunk 1]
```diff
@@ -229,9 +229,12 @@ - (void)setupConversationDataSource
         }
     }
     
-    self.queryController = [self.layerClient queryControllerWithQuery:query];
-    self.queryController.delegate = self;
     NSError *error;
+    self.queryController = [self.layerClient queryControllerWithQuery:query error:&error];
+    if (!self.queryController) {
+        NSLog(@"LayerKit failed to create a query controller with error: %@", error);
```
Comment (chipxsd) [inline Code/Controllers/ATLConversationListViewController.m]: The error condition is not getting handled which makes the behavior after this line undefined.

File: Code/Controllers/ATLConversationListViewController.m [hunk 2]
```diff
@@ -468,9 +471,13 @@ - (BOOL)searchDisplayController:(UISearchDisplayController *)controller shouldRe
             LYRQuery *query = [LYRQuery queryWithQueryableClass:[LYRConversation class]];
             query.predicate = [LYRPredicate predicateWithProperty:@"participants" predicateOperator:LYRPredicateOperatorIsIn value:participantIdentifiers];
             query.sortDescriptors = @[[NSSortDescriptor sortDescriptorWithKey:@"lastMessage.receivedAt" ascending:NO]];
-            self.searchQueryController = [self.layerClient queryControllerWithQuery:query];
             
             NSError *error;
+            self.searchQueryController = [self.layerClient queryControllerWithQuery:query error:&error];
+            if (!self.queryController) {
+                NSLog(@"LayerKit failed to create a query controller with error: %@", error);
```
Comment (chipxsd) [inline Code/Controllers/ATLConversationListViewController.m]: Same as before.

File: Code/Models/ATLConversationDataSource.m [hunk 3]
```diff
@@ -43,11 +43,14 @@ - (id)initWithLayerClient:(LYRClient *)layerClient query:(LYRQuery *)query
         NSUInteger numberOfMessagesAvailable = [layerClient countForQuery:query error:nil];
         NSUInteger numberOfMessagesToDisplay = MIN(numberOfMessagesAvailable, ATLQueryControllerPaginationWindow);
     
-        _queryController = [layerClient queryControllerWithQuery:query];
+        NSError *error = nil;
+        _queryController = [layerClient queryControllerWithQuery:query error:&error];
+        if (!_queryController) {
+            NSLog(@"LayerKit failed to create a query controller with error: %@", error);
```
Comment (chipxsd) [inline Code/Models/ATLConversationDataSource.m]: same

File: Code/Controllers/ATLConversationListViewController.m (hunk 1 above)
Comment (kmahal) [inline Code/Controllers/ATLConversationListViewController.m]: Added a return if nil

File: Code/Controllers/ATLConversationListViewController.m (hunk 2 above)
Comment (kmahal) [inline Code/Controllers/ATLConversationListViewController.m]: We don't handle the error below either.  I can return early.

File: Code/Models/ATLConversationDataSource.m (hunk 3 above)
Comment (kmahal) [inline Code/Models/ATLConversationDataSource.m]: We return an object even if the below execution fails with error, what else should we change here then?

**Pull request merged**